src/jit.rs

```diff
@@ -58,26 +70,41 @@ const R8:  u8 = 8;
 const R9:  u8 = 9;
 const R10: u8 = 10;
 const R11: u8 = 11;
-//const R12: u8 = 12;
+const R12: u8 = 12;
 const R13: u8 = 13;
 const R14: u8 = 14;
 const R15: u8 = 15;
 
+// System V AMD64 ABI
+// Works on: Linux, macOS, BSD and Solaris but not on Windows
+const ARGUMENT_REGISTERS: [u8; 6] = [
+    RDI, RSI, RDX, RCX, R8, R9
+];
+const CALLER_SAVED_REGISTERS: [u8; 7] = [
+    RAX, RCX, RDX, RSI, RDI, R8, R9 //, R10, R11: They aren't mapped and this also conveniently leaves R9 at the end, which is the first argument to be spilled
+];
+const CALLEE_SAVED_REGISTERS: [u8; 6] = [
+    RBP, RBX, R12, R13, R14, R15
+];
+
+// Special registers:
+// RDI Pointer to optional typed return value
+// R10 Stores a constant pointer to mem
+// R11 Scratch register for offsetting
+
 const REGISTER_MAP_SIZE: usize = 11;
```

Comment:
Probably don't need this, but if you want to keep it should apply the same technique to the const register arrays above (`ARGUMETN_REGISTERS`, etc..)